frontend/appflowy_tauri/src/appflowy_app/components/document/TextBlock/TextBlock.hooks.ts

```diff
@@ -10,65 +10,76 @@ import {
   indentNodeThunk,
   splitNodeThunk,
 } from '@/appflowy_app/stores/reducers/document/async_actions';
-import { TextSelection } from '@/appflowy_app/stores/reducers/document/slice';
+import { documentActions, TextSelection } from '@/appflowy_app/stores/reducers/document/slice';
 
-export function useTextBlock(id: string, delta: TextDelta[]) {
-  const { editor, onSelectionChange } = useTextInput(id, delta);
-  const [value, setValue] = useState<Descendant[]>([]);
+export function useTextBlock(id: string) {
+  const { editor, onChange, value } = useTextInput(id);
   const { onTab, onBackSpace, onEnter } = useActions(id);
-  const onChange = useCallback(
-    (e: Descendant[]) => {
-      setValue(e);
-      editor.operations.forEach((op) => {
-        if (op.type === 'set_selection') {
-          onSelectionChange(op.newProperties as TextSelection);
-        }
-      });
-    },
-    [editor]
-  );
+  const dispatch = useAppDispatch();
 
-  const onKeyDownCapture = (event: React.KeyboardEvent<HTMLDivElement>) => {
-    switch (event.key) {
-      case 'Enter': {
-        if (!editor.selection) return;
-        event.stopPropagation();
-        event.preventDefault();
-        const retainRange = getRetainRangeBy(editor);
-        const retain = getDelta(editor, retainRange);
-        const insertRange = getInsertRangeBy(editor);
-        const insert = getDelta(editor, insertRange);
-        void (async () => {
-          await onEnter(retain, insert);
-        })();
-        return;
-      }
-      case 'Backspace': {
-        if (!editor.selection) return;
+  const keepSelection = useCallback(() => {
+    // This is a hack to make sure the selection is updated after next render
+    // It will save the selection to the store, and the selection will be restored
+    if (!editor.selection || !editor.selection.anchor || !editor.selection.focus) return;
+    const { anchor, focus } = editor.selection;
+    const selection = { anchor, focus } as TextSelection;
+    dispatch(documentActions.setTextSelection({ blockId: id, selection }));
+  }, [editor]);
 
-        const { anchor } = editor.selection;
-        const isCollapsed = Range.isCollapsed(editor.selection);
-        if (isCollapsed && anchor.offset === 0 && anchor.path.toString() === '0,0') {
+  const onKeyDownCapture = useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      switch (event.key) {
+        // It should be handled when `Enter` is pressed
+        case 'Enter': {
```

Comment:
Just curious, will the 'shift+enter' keystroke trigger a response in this branch as well?

Reply:
Yes, if we want the 'shift+enter' to trigger another action, we can filter the `shift` key